dspace-api/src/main/java/org/dspace/uclouvain/citations/UCLouvainCitationsServiceImpl.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.uclouvain.citations;
+
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import javax.annotation.PostConstruct;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.content.Item;
+import org.dspace.content.crosswalk.CrosswalkMode;
+import org.dspace.content.crosswalk.StreamDisseminationCrosswalk;
+import org.dspace.content.integration.crosswalks.StreamDisseminationCrosswalkMapper;
+import org.dspace.content.integration.crosswalks.service.ItemExportFormat;
+import org.dspace.content.integration.crosswalks.service.ItemExportFormatService;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.utils.DSpace;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Main service to generate text citations for an item.
+ * We use this instead of the classic DSpace system which is using processes and returns files to the user.
+ * 
+ * @author Michaël Pourbaix (michael.pourbaix@uclouvain.be)
+ */
+public class UCLouvainCitationsServiceImpl implements UCLouvainCitationsService {
+    private Logger logger = LogManager.getLogger(UCLouvainCitationsServiceImpl.class);
+    private StreamDisseminationCrosswalkMapper crosswalkMapper;
+    private static final String CSL_ERROR_REGEX = "\\[CSL STYLE ERROR: .*\\]";
+
+    @Autowired
+    ItemExportFormatService itemExportFormatService;
+
+    @Autowired
+    ItemService itemService;
+
+    @PostConstruct
+    void init() {
+        this.crosswalkMapper =
+            new DSpace().getSingletonService(StreamDisseminationCrosswalkMapper.class);
+    }
+
+    public boolean isValidFormat(Context context, String citationFormat) {
+        return this.crosswalkMapper.getTypes().contains(citationFormat);
+    }
+
+    /**
+     * Generate a citation for a given format and item.
+     * 
+     * @param context The current DSpace context.
+     * @param item The item to generated a citation for.
+     * @param citationFormat The desired format of the citation.
+     * 
+     */
+    public ItemCitation getCitationForItem(Context context, Item item, String citationFormat)
+        throws UnknownCitationFormatException {
+        // Retrieve the crosswalk for the given format.
+        StreamDisseminationCrosswalk citationCrosswalk = this.crosswalkMapper.getByType(citationFormat);
+        if (citationCrosswalk == null) {
+            throw new UnknownCitationFormatException(
+                "The citation format '" + citationFormat + "' is not configured in the system."
+            );
```

Comment:
the "UnknownCitationFormatException" is suffisant, no need to create a big string to explain :: 
```java
throw new UnknownCitationFormatException(citationFormat);
```